code/gateway/gatewaylib.q

```diff
@@ -92,3 +92,199 @@ getserverscross:{[req;att;besteffort]
+    querytable:update 
+        // if no overlap return empty list
+        timeoverlaps:{[st;et;tc] $[(et<tc 0)|st>tc 1;`timestamp$();($[st<tc 0;tc 0;st];$[et>tc 1;tc 1;et])]}'[starttime;endtime;inftc[;1]]
+            from querytable where serverid in modquery`serverid;
+    querytable:update starttime:timeoverlaps[;0],endtime:timeoverlaps[;1] from querytable where serverid in modquery`serverid; 
+    querytable:enlist[`timeoverlaps]_querytable;
+
+    // filter queries not required
+    querytable:$[i:`instruments in key options;
+        select from querytable where(0=count each inftc)|(0<count each inftc[;1])&(not null each starttime)&0<count each instruments;
+        select from querytable where(0=count each inftc)|(0<count each inftc[;1])&not null each starttime];
+    // if no results
+    if[0=count querytable;'`$"gateway error - no info found for that table name and time range."];
+    // convert serverid atoms into their respective serverid lists
+    querytable:update serverid:{x where{any x in y}[;y]each x}[options`procs;serverid],
+        // get servertype
+        servertype:`${string .gw.servers'[x]`servertype}serverid,
+        // convert procs into procname if striped
+        procs:.gw.servers[;`attributes;`procname]@/:serverid from querytable;
+    querytable:update procs:servertype from querytable where not(last each serverid)in modquery`serverid;
+
+    // check if instruments are contained in stripe mapping
+    segids:exec distinct[segid]except ` from querytable;
+    if[all[i,count segids];
+        if[0 < count missinginstruments[options`instruments];
+            // if a queried instrument is not in stripe mapping, reload stripe mapping to bring up-to-date mapping
+            .ds.getstripemapping[];
+            // create messaging if instrument is not in mapping
+            if[[missing:missinginstruments[options`instruments]; 0 < count missing];
+                .lg.o[`.gw.adjustqueriesstripe;raze("No intra-day data for the following instruments: ",(string missing))];
+                    // exit script for intra-day only queries
+                if[[(count options`instruments) = count missing;not `hdb in exec servertype from querytable];
+                    '`$"gateway error - no intra-day data found for one or more queried instrument(s). See logs for missing instrument(s)."]
+            ];
+        ];
+
+        // routing instruments for striped databases
+        if[all[i,any exec any each procs like/:("tr*";"rdb*") from querytable];
+            hdbquery:select from querytable where servertype = `hdb;
+            stripedquery:select from querytable except hdbquery;
+            stripedquery:update 
+                instruments:?[max max each ((group .ds.subreq)first each inftc) in' instruments;
+                    ((group .ds.subreq)first each inftc) inter' instruments;0N] 
+                            from stripedquery;
+            querytable:stripedquery uj hdbquery];
```

Comment:
I've been attempting to implement logic similar to suggested here but I can't find a way around a 'type error pointing to the update statement which is only triggering when querying one instrument.  Querying multiple instruments appears to work. Anyone know why this may be the case?

Reply:
So it's

```suggestion
            querytable :update 
                instruments:?[max max each ((group .ds.subreq)first each inftc) in' instruments;
                    ((group .ds.subreq)first each inftc) inter' instruments;0N] 
                            from querytable where servertype<>`hdb];
```

right?

and the existing code works for single instrument?

Reply:
Yep that's the solution I imagined would work too but it is unfortunately not.  Existing code does indeed work for single instrument queries, yes

Reply:
Behaviour is the same whether the query is intra-day only or historical and intra-day too I should note (so the same behaviour for both when `hdb is and isn't in the querytable)

Reply:
Solved by creating list for single instrument queries prior to running `adjustqueriesstripe`